File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/tx/MultisigSignatures.java
package org.hyperledger.iroha.android.tx;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/** Collection of multisig signatures attached to a transaction. */
public final class MultisigSignatures {
  private final List<MultisigSignature> signatures;

  private MultisigSignatures(final List<MultisigSignature> signatures) {
    Objects.requireNonNull(signatures, "signatures");
    this.signatures = List.copyOf(signatures);
    if (this.signatures.contains(null)) {
      throw new IllegalArgumentException("signatures must not contain null entries");
    }
  }

  /** Construct a bundle from the provided signatures. */
  public static MultisigSignatures of(final List<MultisigSignature> signatures) {
    return new MultisigSignatures(signatures);
  }

  /** Construct a bundle from the provided signatures. */
  public static MultisigSignatures of(final MultisigSignature... signatures) {
    Objects.requireNonNull(signatures, "signatures");
    final List<MultisigSignature> list = new ArrayList<>(signatures.length);
    for (final MultisigSignature signature : signatures) {
      list.add(signature);
    }
    return new MultisigSignatures(list);
  }

  /** Immutable list of signatures. */
  public List<MultisigSignature> signatures() {
    return signatures;
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/tx/MultisigSignatures.kt
package org.hyperledger.iroha.sdk.tx

class MultisigSignatures private constructor(signatures: List<MultisigSignature>) {

    private val _signatures: List<MultisigSignature>

    init {
        for (signature in signatures) {
            requireNotNull(signature) { "signatures must not contain null entries" }
        }
        _signatures = signatures.toList()
    }

    val signatures: List<MultisigSignature> get() = _signatures

    companion object {
        @JvmStatic
        fun of(signatures: List<MultisigSignature>): MultisigSignatures =
            MultisigSignatures(signatures)

        @JvmStatic
        fun of(vararg signatures: MultisigSignature): MultisigSignatures =
            MultisigSignatures(signatures.toList())
    }
}
